install/kubernetes/istio-auth.yaml

```diff
@@ -1173,6 +1173,26 @@ spec:
         ports:
         - containerPort: 80
         - containerPort: 443
+        readinessProbe:
+          exec:
+            command:
+            - /bin/sh
+            - -c
+            - curl -f http://localhost:15000/clusters && curl -f http://localhost:15000/listeners && curl -f http://localhost:15000/routes
```

Comment:
`curl -f http://localhost:15000/clusters` should be sufficient here to verify envoy is running, though it isn't sufficient to completely avoid 404 if RDS hasn't occurred yet. You'd need to query the /stats endpoint and grep for number of successful RDS attempts.

Reply:
Why not query all three as it is for now?

Reply:
It gives the false impression that it querying multiple paths is meaningful for the purposes of verifying enovy is running AFAICT.

Reply:
Actually, the stats is a very good idea. Much more easier than checking for non empty jsons.. May be we can do the whole thing in one shot? like 
```bash
test -z `curl http://localhost:15000/stats| grep update_success | cut -d ':' -f2| tr -d ' '|uniq|grep 0`
```
which is basically going to get LDS, RDS, CDS stats and check the update_success fields (they are usually like `foo.bar.update_success: 1` - hence the cut -d .. stuff). Then if its all >0, uniq will squeeze the list and should typically have "1" or "1" and "2" on separate lines. But if it any of these *DS are 0, then uniq will have
```1
0
1``` or some line with 0. The final grep pulls this 0 out and if its true, the test -z fails.

This can go in as readiness probe, which will make sure that the stuff runs until pod is ready, and not forever.

Reply:
```bash
test -z $(curl -s http://localhost:15000/stats|grep update_success|cut -d ':' -f2|tr -d ' '|sort -n|uniq|grep -e "^0$")
```
tested and works.. if non zero, it has non zero (fail) return code.. and if all update success are >0, test -z returns true

Reply:
update: don't use the expression above, because we have a cluster out.orig-dst-cluster-tcp that does not use SDS.. hence its update_success field is always 0.

Reply:
Using ` with the yaml files does not work.
The only way is run run the command with sh -c "long command"

Reply:
if possible we should avoid running 8 commands to get a ok/not ok status (for instance 1 awk/python/... can do this; or pilot-agent should do it...). As this is only for startup maybe it's ok though.